client.go

```diff
@@ -127,6 +132,41 @@ func (c *Client) recvVersion() error {
 	return nil
 }
 
+func (c *Client) recv() {
+	sendAll := func(res result) {
+		c.mu.Lock()
+		listeners := make([]chan<- result, len(c.inflight))
+		for _, ch := range c.inflight {
+			listeners = append(listeners, ch)
+		}
+		c.mu.Unlock()
+		for _, ch := range listeners {
+			ch <- res
```

Comment:
does this need to have a timeout, or a be none initally non blocking ? If one listener is busy, the ones behind it will not receve this packet.